### PR TITLE
feat: Proto column feature tests and samples

### PR DESCRIPTION
**Pull request by harshachinta**
This PR adds unit tests, integration tests and samples for Proto column feature.

### CODE REVIEW

File: tests/system/conftest.py [hunk 1]
```diff
@@ -74,6 +74,18 @@ def database_dialect():
...
+@pytest.fixture(scope="session")
+def proto_descriptor_file():
+    import os
+
+    dirname = os.path.dirname(__file__)
+    filename = os.path.join(dirname, "testdata/descriptors.pb")
+    return open(
+        filename,
+        "rb",
+    ).read()
```
Comment (asthamohta) [inline tests/system/conftest.py:86]: Close the file

File: tests/system/conftest.py [hunk 2]
```diff
@@ -193,11 +207,15 @@ def shared_database(shared_instance, database_operation_timeout, database_dialec
...
+        ddl_statements = _helpers.DDL_STATEMENTS
+        if not _helpers.USE_EMULATOR:
+            ddl_statements = ddl_statements + _helpers.PROTO_COLUMNS_DDL_STATEMENTS
         database = shared_instance.database(
             database_name,
-            ddl_statements=_helpers.DDL_STATEMENTS,
+            ddl_statements=ddl_statements,
             pool=pool,
             database_dialect=database_dialect,
+            proto_descriptors=proto_descriptor_file,
```
Comment (asthamohta) [inline tests/system/conftest.py:218]: Emulator doesn't support this

File: tests/system/test_database_api.py [hunk 3]
```diff
@@ -562,3 +563,214 @@ def _unit_of_work(transaction, name):
...
+
+    databases_to_delete.append(proto_cols_database)
+
+    proto_cols_database.reload()
+    assert proto_cols_database.proto_descriptors is not None
+    assert any("PROTO BUNDLE" in stmt for stmt in proto_cols_database.ddl_statements)
+
+
+def test_compatibility_of_proto_columns(not_emulator, not_postgres, shared_database):
+    singer_info = singer_pb2.SingerInfo()
```
Comment (asthamohta) [inline tests/system/test_database_api.py:593]: move to test_sessions_api

File: tests/system/conftest.py (hunk 1 above)
Comment (harshachinta) [reply tests/system/conftest.py]: Done

File: tests/system/test_database_api.py (hunk 3 above)
Comment (harshachinta) [reply tests/system/test_database_api.py]: Done

File: tests/system/conftest.py (hunk 2 above)
Comment (harshachinta) [reply tests/system/conftest.py:214]: Emulator support for proto columns is in progress and should be available before mainline merge.

**Review by asthamohta — APPROVED** [review]

File: samples/samples/snippets_test.py [hunk 4]
```diff
@@ -184,8 +193,8 @@ def test_create_database_with_encryption_config(
     assert kms_key_name in out
 
 
-def test_create_database_with_proto_descriptors(capsys, instance_id, database_id):
-    snippets.create_database_with_proto_descriptors(instance_id, database_id)
+def test_create_database_with_proto_descriptor(capsys, instance_id, database_id):
```
Comment (gauravpurohit06) [inline samples/samples/snippets_test.py:196]: Where are we writing test for `test_update_database_with_proto_descriptors `?

**Review by gauravpurohit06 — CHANGES_REQUESTED** [review]
LGTM, with a minor question?

File: samples/samples/snippets_test.py (hunk 4 above)
Comment (harshachinta) [reply samples/samples/snippets_test.py:196]: added

File: samples/samples/snippets.py [hunk 5]
```diff
@@ -2601,73 +2639,102 @@ def read_proto_columns_data(instance_id, database_id):
...
 
 
 # [END spanner_read_proto_columns_data]
 
 
-# [START spanner_read_proto_columns_data_using_helper_method]
-def read_proto_columns_data_using_helper_method(instance_id, database_id):
-    """Reads sample proto column data from the database."""
+# [START spanner_read_proto_columns_data_using_dml]
+def read_proto_columns_data_using_dml(instance_id, database_id):
```
[the remark this answers is not in the document]
Comment (harshachinta) [reply samples/samples/snippets.py]: done

**Review by gauravpurohit06 — APPROVED** [review]

**Pull request merged**